source/shared/cpp/ObjectModel/BaseCardElement.cpp

```diff
@@ -95,15 +78,9 @@ const CardElementType BaseCardElement::GetElementType() const
     return m_type;
 }
 
-std::string BaseCardElement::Serialize() const
-{
-    Json::FastWriter writer;
-    return writer.write(SerializeToJsonValue());
-}
-
 Json::Value BaseCardElement::SerializeToJsonValue() const
 {
-    Json::Value root = GetAdditionalProperties();
+    Json::Value root = BaseElement::SerializeToJsonValue();
 
     root[AdaptiveCardSchemaKeyToString(AdaptiveCardSchemaKey::Type)] = GetElementTypeString();
```

Comment:
Element type string would work. I originally skipped it because element type is... typed. :)

---
In reply to: [253133611](https://github.com/Microsoft/AdaptiveCards/pull/2351#discussion_r253133611) [](ancestors = 253133611)

Reply:
yes :)

---
In reply to: [253133611](https://github.com/Microsoft/AdaptiveCards/pull/2351#discussion_r253133611) [](ancestors = 253133611)